src/containers/SarcophagusCreate/index.tsx

```diff
@@ -0,0 +1,190 @@
+import { useSarcophagiStore } from "../../stores/Sarcophagi";
+import { SarcophagusStore } from "../../stores/Sarcophagi/sarcophagi.interfaces";
+import { useWeb3 } from "../../web3";
+import useFileEncryption from "../../components/SarcophagusCreate/hooks/useFileEncryption";
+import { BigNumber, utils } from "ethers";
+import { SarcophagusCreateValues } from "../../components/SarcophagusCreate/sarcophagusCreate.interfaces";
+import { connect } from "../../web3/providers";
+import CreateTitleAndDescription from "../../components/SarcophagusCreate/CreateTitleAndDescription";
+import Settings from "../../components/SarcophagusCreate/Settings";
+import UploadSarcophagusFile from "../../components/SarcophagusCreate/UploadSarcophagusFile";
+import SelectResurrectionTime from "../../components/SarcophagusCreate/SelectResurrectionTime";
+import ArchaeologistSelectForm from "../../components/SarcophagusCreate/SelectArchaeologistForm";
+import Button from "../../components/layout/Button";
+
+const CreateSarcophagus = () => {
+  const [buttonText, setButtonText] = useState("");
+  const [storageFee, setStorageFee] = useState<number | string | BigNumber | null>(null);
+  const [selectedArchaeologist, setSelectedArchaeologist] = useState<Archaeologist | null>(null);
+  const history = useHistory();
+
+  const {
+    file,
+    setFile,
+    setRecipientPublicKey,
+    setArchaeologistAddress,
+    doubleEncryptedFile,
+    assetDoubleHash,
+  } = useFileEncryption();
+
+  const sarcophagiStore: SarcophagusStore = useSarcophagiStore();
+  const archaeologistsStore: ArchaeologistsStore = useArchaeologistsStore();
+  const { account } = useWeb3();
+  const { approved, approveTransaction } = useApproval();
+
+  const handleArchaeologistSelect = (
+    selectedArchaeologist: Archaeologist,
+    storageFee: number | string | BigNumber
+  ) => {
+    setStorageFee(storageFee);
+    setSelectedArchaeologist(selectedArchaeologist);
+    setArchaeologistAddress(selectedArchaeologist?.currentPublicKey);
+  };
+
+  const successRefresh = () => {
+    sarcophagiStore.loadSarcophagi();
+    archaeologistsStore.loadArchaeologists();
+  };
+
+  const redirect = () => {
+    history.push("/tomb");
+  };
+
+  const createSarcophagus = async (values: SarcophagusCreateValues) => {
+    if (!selectedArchaeologist || !storageFee || !assetDoubleHash || !doubleEncryptedFile) {
+      console.error("🚀 ~ Create values not found");
+      return;
+    }
+    try {
+      const { bounty, diggingFee, recipientPublicKey, resurrectionTime, name } = values;
+      let resurrectionTimeBN: BigNumber = BigNumber.from(Number(resurrectionTime) / 1000);
+      
+      const diggingFeeBN = utils.parseEther(diggingFee.toString());
+      const bountyBN = utils.parseEther(bounty.toString());
+      const storageFeeBN = utils.parseEther(storageFee.toString());
+      let formatedPublicKey;
+      if (recipientPublicKey.substr(0, 4) !== "0x04") formatedPublicKey = "0x04" + recipientPublicKey;
```

Comment:
This`0x04` prefix is handled elsewhere in the same manner. Could refactor this into a utils formatter function so you could just write:

```
const recipientPublicKeyBA = utils.arrayify(formatPublicKey(recipientPublicKey)).slice(1);
```